go/vt/vtgate/vstream_manager.go

```diff
@@ -688,16 +688,10 @@ func (vs *vstream) streamFromTablet(ctx context.Context, sgtid *binlogdatapb.Sha
 			for i, event := range events {
 				switch event.Type {
 				case binlogdatapb.VEventType_FIELD:
-					// Update table names and send.
-					// If we're streaming from multiple keyspaces, this will disambiguate
-					// duplicate table names.
-					ev := event.CloneVT()
-					ev.FieldEvent.TableName = sgtid.Keyspace + "." + ev.FieldEvent.TableName
+					ev := maybeUpdateTableName(event, sgtid.Keyspace, vs.flags.ExcludeKeyspaceFromTableName, extractFieldTableName)
```

Comment:
Note: we could add a validation check to prevent the `ExcludeKeyspaceFromTableName` flag from being set if there are multiple keyspaces being streamed. This could act as a safeguard.

However, some clients may want the performance gain and are willing track client-side which table belongs to which keyspace. So the validation check would not be desirable in this case.

Open to either option on this.